code/modules/projectiles/projectile.dm

```diff
@@ -673,6 +673,17 @@
 
 	return dmg_total > 0 ? (dmg_remaining / dmg_total) : 0
 
+/obj/item/projectile/get_matter()
+	. = matter?.Copy()
+	if(isnull(.)) // empty bullets have no need for matter handling
+		return
+	if(istype(loc, /obj/item/ammo_casing)) // if this is part of a stack
+		var/obj/item/ammo_casing/case = loc
+		if(case.amount > 1) // if there is only one, there is no need to multiply
+			for(var/mattertype in .)
+				.[mattertype] *= case.amount
```

Comment:
Is it normal that when getting the matter a single projectile, the returned matter is multiplied by the number of projectiles in the stack? Or is gun code so cursed that it is an expected behaviour to have the matter of the whole stack returned?

Reply:
the single projectile is the only projectile in the entire stack, and must be treated as the entire stack to allow lathe code to know how much matter is "in" the stack of casings, as otherwise the "virtual projectiles", which only exist hypothetically, would be ignored.